matrix/matrix.py

```diff
@@ -0,0 +1,193 @@
+class Matrix:
+
+    matrix: tuple = ()
+
+    def __post_init__(self):
+        if self.is_valid_matrix():
+            object.__setattr__(self, "matrix", self.matrix)
+            object.__setattr__(self, "order", len(self.matrix))
+            object.__setattr__(self, "row_pointer", 0)
+
+    def __str__(self):
+        return "%s" % (self.matrix,)
+
+    def __repr__(self):
+        return "Matrix(%s)" % (self.matrix,)
+
+    def __neg__(self):
+        return Matrix(tuple(tuple(-item for item in self.matrix[j]) for j in range(self.order)))
+
+    def __add__(self, other):
+        if type(other) is not Matrix:
+            raise ObjectTypeNotMatrix(other)
+
+        if self.order >= other.order:
+            return Matrix(tuple(tuple(self.matrix[j][i] + other.matrix[j][i]
+                                      if i < other.order and j < other.order else self.matrix[j][i]
+                                      for i in range(self.order)) for j in range(self.order)))
+        else:
+            return Matrix(tuple(tuple(self.matrix[j][i] + other.matrix[j][i]
+                                      if i < self.order and j < self.order else other.matrix[j][i]
+                                      for i in (range(other.order))) for j in (range(other.order))))
+
+    def __mul__(self, other):
+        if type(other) is int or type(other) is float:
+            return Matrix(tuple(tuple(self.matrix[j][i] * other
+                                      for i in range(self.order)) for j in range(self.order)))
+
+        elif type(other) is Matrix:
+            if self.order != other.order:
+                raise MatrixNotLegalToMultiply(other)
+
+            return Matrix(tuple(tuple(sum(row[i] * cul[i] for i in range(self.order)) for cul in zip(*other))
+                                for row in self))
+        else:
+            raise ObjectTypeNotLegalToMultiply(other)
+
+    def __rmul__(self, other):
+        if type(other) is not int and type(other) is not float:
+            raise ObjectTypeNotLegalToMultiply(other)
+
+        return Matrix(tuple(tuple(self.matrix[j][i] * other
+                                  for i in range(self.order)) for j in range(self.order)))
+
+    def __sub__(self, other):
+        if type(other) is not Matrix:
+            raise ObjectTypeNotMatrix(other)
+
+        return self + (-other)
+
+    def __truediv__(self, other):
+        if type(other) is not int and type(other) is not float:
+            raise ObjectTypeNotLegalToMultiply(other)
+
+        return Matrix(tuple(tuple(self.matrix[j][i] / other
+                                  for i in range(self.order)) for j in range(self.order)))
+
+    def __eq__(self, other):
+        if type(other) is not Matrix:
+            raise ObjectTypeNotMatrix(other)
+
+        if self.order != other.order:
+            return False
+
+        for i in range(self.order):
+            if self.matrix[i] != other.matrix[i]:
+                return False
+
```

Comment:
you can just check if the tuples are the same, the eq of tuples do what you did now